ebos/eclproblem.hh

```diff
@@ -1100,11 +1091,19 @@ public:
         // used when ROCKCOMP is activated
         const bool invalidateFromMaxWaterSat = updateMaxWaterSaturation_();
         const bool invalidateFromMinPressure = updateMinPressure_();
-        invalidateIntensiveQuantities = invalidateFromMaxWaterSat || invalidateFromMinPressure;
 
+        // update hysteresis and max oil saturation used in vappars
+        const bool invalidateFromHyst = updateHysteresis_();
+        const bool invalidateFromMaxOilSat = updateMaxOilSaturation_();
+
+        // the derivatives may have change
+        bool invalidateIntensiveQuantities = invalidateFromMaxWaterSat || invalidateFromMinPressure || invalidateFromHyst || invalidateFromMaxOilSat;
         if (invalidateIntensiveQuantities)
             this->model().invalidateAndUpdateIntensiveQuantities(/*timeIdx=*/0);
 
+        if (getPropValue<TypeTag, Properties::EnablePolymer>())
+            updateMaxPolymerAdsorption_();
```

Comment:
Should not this also possibly trigger an update of the intensive quantities?

Reply:
It never triggered it before. Lets do the moving first. Then think about when we need to update the intensive quantities.

Reply:
I'll accept that. Just thought this would also be a bugfix?

Reply:
> I'll accept that. Just thought this would also be a bugfix?
I don't know. Depends on what updateMaxPolymerAdsorption_ does.